[Backport 2.3] #11825: Generate new FormKey and replace for oldRequestParams Wishlist

osrecio opened the pull request:
Generate new FormKey afterLogin and set to the beforeRequest(Wishlist)

### Description
Generate new FormKey afterLogin and set to the beforeRequest(Wishlist)

### Fixed Issues (if relevant)
1. magento/magento2#11825: 2.1.9 Item not added to the Wishlist if the user is not logged at the moment he click on the button to add it
2. magento/magento2#11908: Adding to wishlist doesn't work when not logged in

### Manual testing scenarios
1. Create an user account.
2. Logout from the user account
3. Add a product to your Wishlist , you will get redirected to the login page
4. Login

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

osrecio committed:
Generate new FormKey and replace for oldRequestParams

vkublytskyi commented:
@osrecio tests failed as this fix introduces dependency for PageCache on Customer module. We should avoid this for bugfixes. Also we should avoid this as Customer module already has a dependency on PageCashe module and we must avoid circular dependencies. As a solution, I may suggest make a plugin on \Magento\PageCache\Observer\FlushFormKey in a customer module

osrecio commented:
Hi @vkublytskyi I added a Plugin in Customer Module to make the same logic.

vkublytskyi commented on app/code/Magento/Customer/Model/Plugin/CustomerFlushFormKey.php:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * @param FlushFormKey $subject
+     * @param callable $proceed
+     * @param $args
```
Please add `@SuppressWarnings(PHPMD.UnusedFormalParameter)` to avoid static tests failure

osrecio committed:
Add Test CustomerFlushFormKey Plugin

osrecio commented:
@vkublytskyi @ihor-sviziev 
1. Added `@SuppressWarnings(PHPMD.UnusedFormalParameter)` to avoid static tests failure
2. Deleted not related changes
3. Added Unit Tests

vkublytskyi reviewed (changes requested):
Unit test should be updated to fix issue with passed invalid argument. See https://github.com/magento/magento2/pull/12038#discussion_r153482897

Commit:
MAGETWO-83289: [Backport 2.3] #11825: Generate new FormKey and replace for oldRequestParams Wishlist #12042

 - fixed unit test

osrecio commented:
Thanks @vkublytskyi I was going to work on the changes this weekend. Busy Week
